[BUG] compile error on i586: expected `()`, found `FPUControlWord`

## Description

When compiling this crate in GitHub actions on a `i586-unknown-linux-gnu` target, the following error occurs

```txt
   Compiling lexical-parse-float v1.0.5
error[E0308]: mismatched types
  --> /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/lexical-parse-float-1.0.5/src/number.rs:77:23
   |
77 |         let _cw: () = set_precision::<F>();
   |                  --   ^^^^^^^^^^^^^^^^^^^^ expected `()`, found `FPUControlWord`
   |                  |
   |                  expected due to this

For more information about this error, try `rustc --explain E0308`.
error: could not compile `lexical-parse-float` (lib) due to 1 previous error
```

https://github.com/Alexhuszagh/rust-lexical/blob/5af823ff8643527057746153841777449b03d3ca/lexical-parse-float/src/number.rs#L77

## Prerequisites

- Rust version: `nightly-x86_64-unknown-linux-gnu unchanged - rustc 1.86.0-nightly (bef3c3b01 2025-02-04)`
- lexical version: `lexical-parse-float v1.0.5` (`lexical-util` is `v1.0.6`)
- lexical compilation features used: (default?)


## Additional Context
The link to the GitHub CI that fails:

https://github.com/jvdd/argminmax/actions/runs/13158734781/job/36726072805

--- a/.github/workflows/Cross.yml
+++ b/.github/workflows/Cross.yml
@@ -24,6 +24,8 @@ jobs:
           - aarch64-unknown-linux-gnu
           - arm-unknown-linux-gnueabi
           - armv7-unknown-linux-gnueabihf
+          # NOTE: Required for issue #218
+          - i586-unknown-linux-gnu
           - i686-unknown-linux-gnu
           - i686-unknown-linux-musl
           # NOTE: This fails on cross v0.2.1, which is unusual since

--- a/CHANGELOG
+++ b/CHANGELOG
@@ -31,6 +31,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Patch doctests with TODOs in dependencies (#222).
 - CI with OSS-Fuzz for the deprecated `actions/upload-artifact@v3` (#221).
+- Float parsing on `i586` targets (#219).
 - Bug where the `radix` feature wasn't enabling `power-of-two` in `lexical-core` or `lexical` (#204).
 - Fixed performance issues due to a lack of inlining on the Eisel-Lemire algorithm (#210).
 - Issue with parsing non-decimal exponent radixes when using a decimal mantissa radix for floating-point numbers (#212).

--- a/lexical-parse-float/src/number.rs
+++ b/lexical-parse-float/src/number.rs
@@ -74,7 +74,7 @@ impl Number<'_> {
         // function takes care of setting the precision on architectures which
         // require setting it by changing the global state (like the control word of the
         // x87 FPU).
-        let _cw: () = set_precision::<F>();
+        let _cw = set_precision::<F>();
 
         if self.is_fast_path::<F, FORMAT>() {
             let radix = format.radix();

